services/dsp/src/public/js/dsp.js

```diff
@@ -1,10 +1,10 @@
 let dsp = document.currentScript.getAttribute('dsp');
+let bucket = document.currentScript.getAttribute('bucket');
+let cloudEnv = document.currentScript.getAttribute('cloudenv');
 window.addEventListener('load', (event) => {
-  let iframeAws = document.createElement('iframe');
-  let iframeGcp = document.createElement('iframe');
-  // let dsp = document.currentScript.getAttribute('dsp');
-  iframeAws.src = `https://${dsp}/private-aggregation-aws`;
-  iframeGcp.src = `https://${dsp}/private-aggregation-gcp`;
-  document.body.appendChild(iframeAws);
-  document.body.appendChild(iframeGcp);
+  if ((cloudEnv == 'aws' || cloudEnv == 'gcp') && bucket != 'undefined') {
```

Comment:
Just to confirm the logic : if the "bucket" query string is not set, the private-aggregation script won't be loaded. is that intended behavior ? 

in the services/dsp/src/public/js/private-aggregation-worklet.js you are setting a default bucketKey if it's not found in Shared Storage, but I wonder if the SS code will be executed if we don't meet the condition "bucket != 'undefined'" here.

Reply:
Hi Severin,

Yup. It was supposed to be that if bucket key is not set in the url, we will pre-set the bucket key for the report. I've removed the `bucket!='undefined'` requirement.

Thanks!

Reply:
Thanks Maybelline. Looks good to me.